### ISSUE TITLE
CPU stuck at 400MHz when thermald is enabled on boot.

### ISSUE THREAD

**Issue by depereo** [intel/thermal_daemon#291]
Hi,

I've run into a similar issue first reported in #280.
```
Laptop:~$ thermald --version
2.4.1
Laptop:~$ uname -r
5.10.11-200.fc33.x86_64
```

I've been unable to replicate this issue 'manually' when running `thermald --no-daemon --loglevel=info --adaptive` as described in #280; it seems to only 'trigger' if thermald is enabled by systemd and the laptop is rebooted.

Restarting thermald (`systemctl restart thermald`) after logging in 'unlocks' the cpu for a while, but it'll go back to locking all cores to 400MHz after a few minutes, requiring additional restarts of the service.

If the cpu is 'locked' to 400MHz, and thermald is stopped, the cpu remains at 400MHz indefinitely.
If the cpu is in an 'unlocked' state, and thermald is stopped, the cpu stays at a variable frequency range and operates as expected.

service file:
```
Laptop:~$ cat /lib/systemd/system/thermald.service 
[Unit]
Description=Thermal Daemon Service
ConditionVirtualization=no

[Service]
Type=dbus
SuccessExitStatus=2
BusName=org.freedesktop.thermald
ExecStart=/usr/sbin/thermald --systemd --dbus-enable --adaptive
Restart=on-failure

[Install]
WantedBy=multi-user.target
Alias=dbus-org.freedesktop.thermald.service
```

Swapping between various kernel versions and bios versions has had no effect.

CPU: Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz

**Comment by spandruvada** [comment]
Change the systemd service file to enable logging

ExecStart=/usr/sbin/thermald --systemd --dbus-enable --adaptive --loglevel=debug

Then save

And 
systemctl daemon-reload

The reboot and collect logs from journalctl -rb

**Comment by depereo** [comment]
I've made those changes. Logs uploaded here include booting, experiencing the 'lock' to 400MHz, restarting to 'unlock' the cpu and re-experiencing the 'lock'.

`journalctl -rb | grep thermald` output:


[output.txt](https://github.com/intel/thermal_daemon/files/5907872/output.txt)

**Comment by spandruvada** [comment]
Can you build thermald yourself and try this change?

[issue_291.zip](https://github.com/intel/thermal_daemon/files/5908210/issue_291.zip)

You can apply with
git apply issue_r91.diff

Then build and install.

**Comment by depereo** [comment]
The build, with this diff applied, works well and doesn't seem to trigger the 400MHz locking condition.

Thanks for the fast fix, it's appreciated.

**Comment by Tiller** [comment]
Hi, I was about to post a similar issue with the logs as asked in #280 but I stumbled upon this issue first. I'm currently trying the patch and it also seems to work fine on my end. I'll update you if it doesn't.
In case you want to investigate my logs to be sure it was the same issue, here are the logs : https://stuff.stooit.com/d/1/6019033dd18ef/beforepatch.tar.gz (i7-8665U randomly stuck at 400MHz for a time, with thermald's master from a week ago and kernel 5.8.0-38-generic)

**Comment by spandruvada** [comment]
Thanks for the test. It will be great if you can send me logs with
patch applied also, I want to check if the issue is fixed.

Thanks!

**Comment by spandruvada** [comment]
Also what is make and model of this system?

**Comment by spandruvada** [comment]
I pushed the change. Please checkout and confirm before I release a new version.
Please send me logs after this change applied, so that I can confirm that patch is working.

**Comment by depereo** [comment]
new log file with patched version, no locking experienced:
[output.txt](https://github.com/intel/thermal_daemon/files/5914120/output.txt)

System details: Dell Latitude 7400

I've done a 'burn-in' test overnight with the patch applied and the locking has not occurred - previously the system spent most of its time locked to 400MHz.

**Comment by depereo** [comment]
Checked out https://github.com/intel/thermal_daemon/commit/03c01f0883c9a75e918e747ce2b9f674d8e7bf5c and rebuilt, logs from that version attached as well here:

No observed issues, CPU frequency fluctuating properly with workload.

[output.txt](https://github.com/intel/thermal_daemon/files/5914240/output.txt)

**Comment by spandruvada** [comment]
Logs looks good. It is not throttled by TMEM sensor at 40C.

**Comment by Tiller** [comment]
For reference I also have a Dell Latitude 7400!

**Comment by spandruvada** [comment]
Added fix v2.4.2
